### ISSUE TITLE
Color issues with vncviewer and X11 forwarding between Linux on s390x and x86-64

### ISSUE THREAD

**Issue by mhillenbrand** [TigerVNC/tigervnc#1252]
**Describe the bug**
When using vncviewer with X11 forwarding via SSH such that vncviewer runs on Linux on s390x and is displayed on an X11 server on Linux on x86_64, pixels have the wrong color. There are what looks like two different modes in how pixel values can turn out wrong. On the side of the Xvnc server, pixel values are ok (tested with vncviewer without X forwarding).

See screenshot below

I have tested the following scenarios for comparison, all of which work as expected:
- vncviewer running on my Linux laptop, VNC via SSH tunnel to Xvnc server on s390x
- vncviewer connecting to Xvnc server on the same s390x host, displaying into another Xvnc session
- vncviewer and Xvnc server both on s390, displaying via X11 forwarding in another Xvnc session on a different Linux s390x host

**To Reproduce**
1. From a Linux x86 system, connect to a Linux system on s390x via SSH with X forwarding
2. Start a Xvnc session on that Linux system
```
tigervncserver :0 -geometry 800x600 -depth 32
```
3. Start vncviewer on the same Linux system
```
xtigervncviewer :0
```
4. On vncviewer's window on your origin X session, observe incorrect colors.
5. Optionally, start X11 programs such as `xlogo`, `xclock`, or `xeyes` to make the problem obvious
6. Optionally, connect to the vnc session directly (i.e., with a vnc client on the origin system) to compare

**Expected behavior**
The combination of X11 forwarding for vncviewer between different platforms is arguably not very common. Yet, I still think a user should see the remote screen with correct colors even in that scenario.

**Screenshots**
- Broken display of vncviewer with X11 forwarding
![image](https://user-images.githubusercontent.com/1150167/118849233-d1a62600-b8cf-11eb-8aaa-fe904a52657c.png)
- comparing vncviewer with X11 forwarding (bottom right) with direct VNC connections (vinagre in surrounding window and _nested_ tiger vnc viewer)
![image](https://user-images.githubusercontent.com/1150167/118856791-9dcefe80-b8d7-11eb-91a1-5262293e004f.png)

**Client (please complete the following information):**
 - OS: Ubuntu 21.04, Red Hat Enterprise Linux 8.3
 - VNC client: TigerVNC
 - VNC client version: v1.10.1
 - Client downloaded from: Linux distribution

**Server (please complete the following information):**
 - OS: Ubuntu 21.04, Red Hat Enterprise Linux 8.3
 - VNC server: TigerVNC
 - VNC server version: v1.10.1
 - Server downloaded from: Linux distribution
 - Server was started using: `tigervncserver :0 -geometry 800x600 -depth 32 -localhost no`

**Additional context**
Both client and server running on s390x. X11 forwarding to a Linux on x86-64

**Comment by CendioOssman** [comment]
This has been fixed in #1084 and will be included in the next release. Until then you can try a nightly build or see if the distributions can include a backported fix to their packaging.

**Comment by mhillenbrand** [comment]
Thanks!
